### ISSUE TITLE
undefined reference to `__atan2f_finite' Ubuntu 20.04

### ISSUE THREAD

**Issue by femust** [intel-isl/Open3D#1909]
**IMPORTANT: Please use the following template to report the bug.**
_____

**Describe the bug**
Ubuntu 20.04, problem with:

```
/usr/bin/ld: ../../lib/libOpen3D.so: undefined reference to `__atan2f_finite'
/usr/bin/ld: ../../lib/libOpen3D.so: undefined reference to `__asinf_finite'
```
probably related to: https://www.pgroup.com/userforum/viewtopic.php?t=7648

**To Reproduce**

install ubuntu 20.04
compile

**Expected behavior**
compiled version

**Environment (please complete the following information):**

- Operating system: Ubuntu 20.04
- Python version: Python 3.8
- Open3D version: 0.10
- Is this remote workstation?: yes 
- How did you install Open3D?:  build from source
- Compiler version (if built from source): clang10.0

**Comment by yxlao** [comment]
Could you share the cmake config flag?

**Comment by femust** [comment]
actually, since I was struggling and time was important to me, I downgraded to ubuntu 18.04, but I tried lots of things, for example, set up these flags: https://github.com/google/filament/issues/2146

**Comment by kelia** [comment]
experiencing the same issue, any update on this?

**Comment by stefanseefeld** [comment]
and likewise here. Let me know if you need any build logs or other info. I'm building from source, in a Ubuntu 20.04 docker container.

**Comment by christianlandgraf** [comment]
I experienced the same issue but similarly to #2094, I was able to solve it by manually installing filament from source and adding the paths to the CMakeLists in Open3d (Variables FILAMENT_PRECOMPILED_ROOT and FILAMENT_SOURCE_ROOT).

For filament, I followed the instructions from [here](https://github.com/google/filament/blob/main/BUILDING.md#linux) using their dependencies (clang-10, ninja, etc.) and compiled it without java bindings.

**Comment by zangel** [comment]
I also had this issue and the solution is as @femust  pointed out to the filament issue [https://github.com/google/filament/issues/2146](url).
Actually filament needs to be compiled with builtin functions disabled (`-fno-builtin`) (i am attaching a patch for that), then Open3D needs to be build with `-DBUILD_FILAMENT=YES`.
[ext_filament_patch.txt](https://github.com/intel-isl/Open3D/files/5033193/ext_filament_patch.txt)

**Comment by chuong** [comment]
@christianlandgraf I tried your solution and successfully compile filament separately and passed its part but got stuck below when compiling Open3D  with `make`

```
-- Installing: /media/ngu10t/DATA1/repos/Open3D/build/assimp/include/assimp/Compiler/pushpack1.h
-- Installing: /media/ngu10t/DATA1/repos/Open3D/build/assimp/include/assimp/Compiler/poppack1.h
-- Installing: /media/ngu10t/DATA1/repos/Open3D/build/assimp/include/assimp/Compiler/pstdint.h
[ 21%] Completed 'ext_assimp'
[ 21%] Built target ext_assimp
make: *** [Makefile:130: all] Error 2
```
Could you please provide more details of your solution? Thanks.

@zangel I tried your patch/solution but same error reported at the beginning of this thread still occurred. Could you provide a complete reproducible steps? Thanks.

My system is also Ubuntu  20.04 .

**Comment by yxlao** [comment]
Resolved by #2438, closing this issue.

The `-fno-builtin` approach mentioned by @zangel is used. It is also recommended by https://github.com/google/filament/issues/2875.
